controllers/openstackmachine_controller.go

```diff
@@ -423,6 +423,7 @@ func (r *OpenStackMachineReconciler) reconcileMachineState(scope *scope.WithLogg
 
 		// Set properties required by CAPI machine controller
 		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack:///%s", *openStackServer.Status.InstanceID))
+		openStackMachine.Status.InstanceID = openStackServer.Status.InstanceID
```

Comment:
What problem are we trying to solve here? I think we made that on purpose but I'll like you know what your issue is before we revert that change.

Reply:
Hi!
As an operator of CAPI/CAPO, I have a Kubernetes controller that depends on `OpenStackMachine.Stataus.InstanceID`.
And the disappearance of InstanceID will cause a bug.

Are you planning to delete this field in `v1beta1`?
I understand that `InstanceID` can be still retrieved from `ProviderID`, but I hope the breaking change is delivered with the update of API versions like `v1beta2`.

Thank you.

Reply:
I guess since we're v1beta1, we shouldn't have remove that field. I won't block this PR.